### PR TITLE
[SPARK-16936] [SQL] Case Sensitivity Support for Refresh Temp Table

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?

Currently, the `refreshTable` API is always case sensitive.

When users use the view name without the exact case match, the API silently ignores the call. Users might expect the command has been successfully completed. However, when users run the subsequent SQL commands, they might still get the exception, like 

```
Job aborted due to stage failure: 
Task 1 in stage 4.0 failed 1 times, most recent failure: Lost task 1.0 in stage 4.0 (TID 7, localhost): 
java.io.FileNotFoundException: 
File file:/private/var/folders/4b/sgmfldk15js406vk7lw5llzw0000gn/T/spark-bd4b9ea6-9aec-49c5-8f05-01cff426211e/part-r-00000-0c84b915-c032-4f2e-abf5-1d48fdbddf38.snappy.parquet does not exist
```

This PR is to fix the issue.
### How was this patch tested?

Added a test case.

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @rxin @cloud-fan This is a simple fix. Could you review the fix? Thanks!

File: sql/core/src/test/scala/org/apache/spark/sql/MetadataCacheSuite.scala [hunk 1]
```diff
@@ -85,4 +86,28 @@ class MetadataCacheSuite extends QueryTest with SharedSQLContext {
...
+          // Create a Parquet directory
+          spark.range(start = 0, end = 100, step = 1, numPartitions = 3)
+            .write.parquet(location.getAbsolutePath)
+
+          // Read the directory in
+          spark.read.parquet(location.getAbsolutePath).createOrReplaceTempView("view_refresh")
+
+          // Delete a file
+          deleteOneFileInDirectory(location)
+          intercept[SparkException] {sql("select count(*) from view_refresh").first()}
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/MetadataCacheSuite.scala]: nit: use `intercept[..](...)` for one line code.

**Comment by cloud-fan** [comment]
LGTM, is it a regression for 2.0?

**Comment by gatorsmile** [comment]
This is not a regression, but Spark 2.0 has the issue. Thanks!

**Comment by cloud-fan** [comment]
thanks, merging to master and 2.0!

**Pull request closed without merging**